_data-prepper/pipelines/configuration/sources/kafka.md

```diff
@@ -0,0 +1,145 @@
+`serde_format` | No | String | Indicates the serialization and the deserialization format of the messages in the topic. Default is `plaintext`.
+`autocommit` | No | Boolean | When `false`, the consumer's offset will no be periodically committed to Kafka in the background. Default is `false`.
+`commit_interval` | No | Integer | When `autocommit` is set to `true`, sets the frequency in seconds that the consumer offsets are auto-committed to Kafka through Kafka's `auto.commit.interval.ms` option. Default is `5s`.
+`session_timeout` | No | Integer | The timeout time the source detects client failures when using Kafka's group management features, which can be used to balance the data stream. Default is `45s`.
+`auto_offset_reset` | No | String | Automatically resets the offset to the earlier or latest offset through Kafka's `auto.offset.reset` option. Default is `latest`.
+`thread_waiting_time` | No | Integer | The time that threads wait until the preceding thread completes it's task and signals the next thread. The Kafka consumer API poll timeout value is set to half of this setting. Default is `5s`.
+`max_partition_fetch_bytes` | No | Integer | Sets the maximum limit in bytes for much data returns from each partition through Kafka's `max.partition.fetch.bytes`. Default is `1048676`.
+`heart_beat_interval` | No | Integer | The expected time between heartneats to the consumer coordinator when using Kafka's group management facilities through Kafka's `heartbeat.interval.ms` setting. Default is `1s`
+`fetch_max_wait` | No | Integer | The maximum amount of time the server blocks a fetch request when there isn't sufficient data to satisfy the `fetch_min_bytes` requirement through Kafka's `fetch.max.wait.ms` setting. Default is `500`.
+`fetch_max_bytes` | No | Integer | The maximum record size accepted by the broker through Kafka's `fetch.max.bytes` setting. Default is `52428800`.
+`fetch_min_bytes` | No | Integer | The minimum amount of data the server returns during a fetch request through Kafka's `retry.backoff.ms` setting. Default is `1`.
+`retry_backoff` | No | Integer | The amount of time to wait before attempting to retry a failed request to a given topic partition. Default is `10s`.
+`max_poll_interval` | No | Integer | The maximum delay between invocations of a `poll()` when using group management through Kafka's `max.poll.interval.ms` option.  Default is `300s`.
+`consumer_max_poll_records` | No | Integer | The maximum number of records returned in single `poll()` call through Kafka's `max.poll.records` setting. Default is `500`.
+`key_mode` | No | String | Indicates how the key field of Kafka message should be handled. The default setting is `include_as_field`, which includes the key in the `kafka_key` event. `include_as_matedata` places the key in the event's metadata. `discard` discards the key. 
+
+### Schema
+
+Use the following option is required inside the `schema` array.
```

Comment:
schema is not an array. May be you should say "inside the `schema` config option"?